### PR TITLE
Move every files taken from gaia to js/lib

### PR DESCRIPTION
**Pull request by autra**
Hey @fabi1cazenave 

A first pass of refactoring. Please r?

### CODE REVIEW

**Comment by fabi1cazenave** [comment]
Please add a README in this `/lib` directory to describe its contents. Sorry for being so old-school. :-/

Are all these files extracted from Gaia?

**Comment by autra** [comment]
@fabi1cazenave ready for review btw :-)

File: js/lib/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Lib directory
+
+Contains all the js files coming from external repo.
```
Comment (fabi1cazenave) [inline js/lib/README.md]: +s

File: js/lib/config.js [hunk 2]
```diff
@@ -1,3 +1,6 @@
+/**
+ * This file is a copy of shared/js/contacts/import/utilities/config.js in gaia.
+ */
```
Comment (fabi1cazenave) [inline js/lib/config.js]: If this is really a copy, then leave this file alone so it remains a copy, and put that comment in the README please.

File: js/lib/http_rest.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
+/**
+ * This file is a copy of ./shared/js/contacts/utilities/http_rest.js from
+ * gaia.
+ */
```
Comment (fabi1cazenave) [inline js/lib/http_rest.js]: same here

File: js/lib/image_thumbnail.js [hunk 4]
```diff
@@ -1,3 +1,8 @@
+/**
+ * this file is a copy of./shared/js/contacts/utilities/image_thumbnail.js from
+ * gaia
+ */
+
```
Comment (fabi1cazenave) [inline js/lib/image_thumbnail.js]: same here

File: js/lib/image_utils.js [hunk 5]
```diff
@@ -1,4 +1,8 @@
 /**
+ * This file is a copy of ./shared/js/image_utils.js from gaia
+ */
+
+/**
```
Comment (fabi1cazenave) [inline js/lib/image_utils.js]: same here

File: js/lib/misc.js [hunk 6]
```diff
@@ -1,3 +1,8 @@
+/**
+ * This file is a copy of ./shared/js/contacts/import/utilities/misc.js from
+ * gaia
+ */
+
```
Comment (fabi1cazenave) [inline js/lib/misc.js]: same here

File: js/lib/simple_phone_matcher.js [hunk 7]
```diff
@@ -1,3 +1,7 @@
+/**
+ * This file is a copy of ./shared/js/simple_phone_matcher.js from gaia.
+ */
+
```
Comment (fabi1cazenave) [inline js/lib/simple_phone_matcher.js]: same here

File: index.html [hunk 8]
```diff
@@ -31,16 +31,16 @@
...
-    <script src="js/config.js" defer></script>
-    <script src="js/image_utils.js" defer></script>
+    <script src="js/lib/gaia/config.js" defer></script>
+    <script src="js/lib/gaia/image_thumbnail.js" defer></script>
+    <script src="js/lib/gaia/image_utils.js" defer></script>
+    <script src="js/lib/gaia/misc.js" defer></script>
+    <script src="js/lib/gaia/simple_phone_matcher.js" defer></script>
 
     <!-- These has been taken from gaia, but had substantial modifications -->
+    <script src="js/lib/http_rest.js" defer></script>
```
Comment (autra) [inline index.html]: This will have substantial modifications in a next PR.

**Comment by autra** [comment]
@fabi1cazenave please R? again :-)

**Comment by autra** [comment]
Let's timeout this. Don't hesitate to open issues if something is wrong.

**Pull request merged**